tarofreighter/chain_porter.go

```diff
@@ -522,24 +522,48 @@ func createDummyOutput() *wire.TxOut {
 // adjustFundedPsbt takes a PSBT which may have used BIP 69 sorting, and
 // creates a new one with outputs shuffled such that the change output is the
 // last output.
-func adjustFundedPsbt(pkt *psbt.Packet, changeIndex uint32,
-	anchorInputValue int64) {
+func adjustFundedPsbt(pkt *psbt.Packet, changeIndex int32,
+	anchorInputValue int64) int32 {
+
+	// If there is no change there's nothing we need to do.
+	if changeIndex == -1 {
+		return changeIndex
+	}
+
 	// Store the script and value of the change output.
+	maxOutputIndex := len(pkt.UnsignedTx.TxOut) - 1
 	changeOutput := pkt.UnsignedTx.TxOut[changeIndex]
-	changeAmtSats := changeOutput.Value
-	changeScript := changeOutput.PkScript
 
 	// Overwrite the existing change output, and restore in at the
 	// highest-index output.
-	maxOutputIndex := len(pkt.UnsignedTx.TxOut) - 1
 	pkt.UnsignedTx.TxOut[changeIndex] = createDummyOutput()
-	pkt.UnsignedTx.TxOut[maxOutputIndex].PkScript = changeScript
-	pkt.UnsignedTx.TxOut[maxOutputIndex].Value = changeAmtSats
+	pkt.UnsignedTx.TxOut[maxOutputIndex].PkScript = changeOutput.PkScript
```

Comment:
q: what happens with the info that was in the last output? Can we simply overwrite it and forget about it? Was it a placeholder output with 0 value?

Reply:
Yes, there are two dummy outputs being created that we'll overwrite later on anyway.